src/main/java/app/UserConfigurationBuilder.java

```diff
@@ -0,0 +1,60 @@
+package app;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class UserConfigurationBuilder
+{
+    private int port;
+    private String stompConnectionEndpoint;
+    private String brokerDestinationEndpoint;
+    private List<TriggerEndpointConfiguration> triggerEndpointConfigurations;
+
+    private UserConfigurationBuilder()
+    {
+    }
+
+    public static UserConfigurationBuilder builder()
```

Comment:
Why does it need to be static?

Reply:
It's a feature. This is made to limit bare constructor calls from user point of view. As a user of any class you should not be responsible for creating it's objects. I know this may sound over eager and sth like over-interpretation but this is the direction I see modern libraries use. They limit ways user can interact with it's code by providing only selected options. This also allows for much more flexibility when it comes to change. Probably this won't be changed, since `UserConfiguration` won't participate in any inheritance chain, but this is still a reason to me.

Here is [something](https://stackoverflow.com/questions/4029622/how-to-use-static-factory-methods-instead-of-constructors) about that stuff

Reply:
Ok, now I see the point, thanks.